src/inmanta/user_setup.py

```diff
@@ -0,0 +1,155 @@
+
+        value = None
+        while value not in ["yes", "no"]:
+            value = click.prompt("Do you want to generate a new configuration? yes/no")
+
+        if value == "yes":
+            click.echo("Add the following to the configuration in /etc/inmanta/inmanta.d/auth.cfg:\n")
+            click.echo(generate_signing_config())
+
+        raise click.ClickException("Make sure signing configuration is added to the config. See the documentation for details.")
+
+    click.echo(f"Authentication signing config:                          {click.style('enabled', fg='green')}")
+
+    # TODO: verify web-console config (if any)
+
+
+async def get_database_connection() -> asyncpg.Pool:
+    database_host = server_config.db_host.get()
+    database_port = server_config.db_port.get()
+
+    database_username = server_config.db_username.get()
+    database_password = server_config.db_password.get()
+    connection_pool_min_size = server_config.db_connection_pool_min_size.get()
+    connection_pool_max_size = server_config.db_connection_pool_max_size.get()
+    connection_timeout = server_config.db_connection_timeout.get()
+    return await data.connect(
+        database_host,
+        database_port,
+        server_config.db_name.get(),
+        database_username,
+        database_password,
+        connection_pool_min_size=connection_pool_min_size,
+        connection_pool_max_size=connection_pool_max_size,
+        connection_timeout=connection_timeout,
+    )
+
+
+async def do_user_setup() -> None:
+    """Perform the user setup that requires the database interaction"""
+    connection = None
+    try:
+        connection = await get_database_connection()
+        users = await data.User.get_list()
+
+        if len(users):
+            raise click.ClickException(
+                "There are already users in the database. If you want to reset the password, use the --reset option."
+            )
+
+        username = click.prompt("What username do you want to use?", default="admin")
+        password = click.prompt("What password do you want to use?", hide_input=True)
+
+        pw_hash = nacl.pwhash.str(password.encode())
+
+        # insert the user
+        user = data.User(
+            username=username,
+            password=pw_hash.decode(),
+            enabled=True,
+            auth_method="password",
+        )
+        await user.insert()
+
+        click.echo(f"User {username}:                                             {click.style('created', fg='green')}")
+    finally:
+        if connection is not None:
+            await data.disconnect()
+
+    click.echo("Make sure to (re)start the orchestrator to activate all changes.")
+
+
+@click.command(help="Do the initial user setup")
+@click.option("--reset", help="Reset the password to recover a lost password", is_flag=True)
+def cmd(reset: bool) -> None:
+    # validate the setup so that we can setup a new user
+    validate_server_setup()
```

Comment:
What if the database schema is outdated? Do we just expect this command to be run only on an up to date server?

Reply:
I would be in favor of making an attempt to check if the schema is outdated. `inmanta.data.schema` probably has some methods that could help here, but I'm not sure if it requires the server to be running.

I'd ask the question to the team and see if anyone else has an opinion.

Reply:
I mainly would like a clear error message when the schema would be outdated.

Reply:
I would make it broader than that. What if we make a change to the users table, and consequentially to this command? I would be in favor of aborting the command with a clear error message if there are migration scripts that are not applied to the active schema, because this means that the Python source and the database schema are out of sync.

Reply:
Ow I didn't see @sanderr 's message. Could you point me in a direction on how this can be done?

Reply:
`inmanta.data.schema` is that direction. If you need more I'd need to have a look myself, so let me know. But I think we first need to reach consensus. Let's await @arnaudsjs' response first.

Reply:
Fine by me to make the scope broader.